src/test/components/NetworkChart.test.js

```diff
@@ -155,6 +174,58 @@ describe('NetworkChartRowBar phase calculations', function() {
     ]);
   });
 
+  it('displays properly a network marker even when it crosses the boundary', () => {
+    const {
+      dispatch,
+      getPhaseElementStyles,
+      getBarElementStyle,
+    } = setupWithPayload(
+      getNetworkMarkers({
+        uri: 'https://mozilla.org/img/',
+        id: 100,
+        startTime: 10,
+        // With an endTime at 109, the profile's end time is 110, and so the
+        // profile's length is 100, which gives integer values for test results.
+        endTime: 109,
+        payload: {
+          pri: 20,
+          count: 10,
+          domainLookupStart: 20,
+          domainLookupEnd: 24,
+          connectStart: 25,
+          tcpConnectEnd: 26,
+          secureConnectionStart: 26,
+          connectEnd: 28,
+          requestStart: 30,
+          responseStart: 60,
+          responseEnd: 80,
+        },
+      })
+    );
+
+    // Note: "10" here means "20" in the profile, because this is the delta
+    // since the start of the profile (aka zeroAt), and not an absolute value.
+    dispatch(commitRange(10, 50));
+
+    // The width is bigger than the mocked available width (which is 200px) but
+    // this is expected.
+    // It's also expected that the left value is less than TIMELINE_MARGIN_LEFT,
+    // because the range start is after the start of the marker.
+    expect(getBarElementStyle()).toEqual('width: 495px; left: 100px;');
```

Comment:
Will this cause us layout issues when we have gigantic for the width?

Reply:
I didn't see any problem, even performance problem, when trying to zoom in a lot on a marker.
For example: https://deploy-preview-1985--perf-html.netlify.com/public/8b2e92d6b9d9a506737fbed3d0fa5424e3b34a0c/network-chart/?globalTrackOrder=0-1-2-3-4&hiddenGlobalTracks=1-2-3&localTrackOrderByPid=3164-1-2-0~12671-0~12665-0~12605-0~12558-0-1~&range=6.3742_7.3422~6.7724_6.8094~6.7872_6.7892&thread=5&v=3
The grey markers are 5 million pixels wide, and I don't see any problem, at least in Firefox :)

Reply:
By the way we can also definitely revisit this choice if we see this has problems later.